udata/templates/dataset/display.html

```diff
@@ -190,7 +190,7 @@ <h3 class="text-left">{{ _('Producer') }}</h3>
                                 </a>
                                 {% for badge in dataset.badges %}
                                     <a href="{{ url_for('front.search', badge=badge) }}">
-                                        {{ badge }}</a>{% if not loop.last %}, {% endif %}
+                                        {{ dataset.__badges__[badge.kind] }}</a>{% if not loop.last %}, {% endif %}
```

Comment:
Do we want to expose `__private__` in templates? Why iterating over badges is not enough?